### PR TITLE
Fix/binary transpose combine

### PR DESCRIPTION
**Pull request by xhuohai**
fix combine transpose and const binary when const is scalar

### CODE REVIEW

File: src/Nncase.Tests/Rules/Neutral/UnitTestCombineTranspose.cs [hunk 1]
```diff
@@ -30,9 +30,11 @@ public class UnitTestCombineTranspose : TransformTestBase
     {
         // BinaryOp binaryOp, int[] lShape, int[] rShape, int[] perm, bool leftConst
         { BinaryOp.Add, new[] { 1, 32, 32, 64, }, new[] { 64 }, new[] { 0, 3, 1, 2 }, false },
+        { BinaryOp.Add, new[] { 1, 32, 32, 64, }, new[] { 1 }, new[] { 0, 3, 1, 2 }, false },
```
Comment (sunnycase) [inline src/Nncase.Tests/Rules/Neutral/UnitTestCombineTranspose.cs:33]: Why not use `Array.Empty<int>()`.

Comment (xhuohai) [reply src/Nncase.Tests/Rules/Neutral/UnitTestCombineTranspose.cs]: changed.

File: src/Nncase.Tests/Rules/Neutral/UnitTestCombineTranspose.cs [hunk 2]
```diff
@@ -217,10 +219,10 @@ public void TestCombineTransposeConcatNegative(int[] inShape, int[][] perm, int
     public void TestCombineTransposeConstBinaryPositive(BinaryOp binaryOp, int[] lShape, int[] rShape, int[] perm, bool leftConst)
     {
         Expr lhs = leftConst ?
-          Const.FromValue(Random.Normal(DataTypes.Float32, 0, 1, 3, lShape).Evaluate()) :
+          lShape.Length == 0 ? 0.5f : Const.FromValue(Random.Normal(DataTypes.Float32, 0, 1, 3, lShape).Evaluate()) :
```
Comment (sunnycase) [inline src/Nncase.Tests/Rules/Neutral/UnitTestCombineTranspose.cs:222]: Does this line should be reverted?

File: src/Nncase.Tests/Rules/Neutral/UnitTestCombineTranspose.cs [hunk 3]
```diff
@@ -217,10 +219,10 @@ public void TestCombineTransposeConcatNegative(int[] inShape, int[][] perm, int
...
           new Var("lhs", new TensorType(DataTypes.Float32, lShape));
         Expr rhs = leftConst ? new Var("b", new TensorType(DataTypes.Float32, rShape)) :
-          Const.FromValue(Random.Normal(DataTypes.Float32, 0, 1, 4, rShape).Evaluate());
+          rShape.Length == 0 ? 0.2f : Const.FromValue(Random.Normal(DataTypes.Float32, 0, 1, 4, rShape).Evaluate());
```
Comment (sunnycase) [inline src/Nncase.Tests/Rules/Neutral/UnitTestCombineTranspose.cs:225]: Same as above.

File: src/Nncase.Tests/Rules/Neutral/UnitTestCombineTranspose.cs (hunk 2 above)
Comment (xhuohai) [reply src/Nncase.Tests/Rules/Neutral/UnitTestCombineTranspose.cs:222]: I want to generate a scalar if shape is a empty array, shape.Length is not good?

File: src/Nncase.Passes/Rules/Neutral/CombineTranspose.cs [hunk 4]
```diff
@@ -73,6 +73,11 @@ public CombineConstBinaryTranspose()
 
         if (x is Const)
         {
+            if (x.CheckedShape.Rank == 0)
+            {
+                return Transpose(Binary(binary.BinaryOp, x, y), perm);
+            }
```
Comment (aaltonenzhang) [inline src/Nncase.Passes/Rules/Neutral/CombineTranspose.cs:79]: please create Binary using Binary(binary.BinaryOp, x, y).InheritMetaData(binaryCall)

File: src/Nncase.Passes/Rules/Neutral/CombineTranspose.cs [hunk 5]
```diff
@@ -89,6 +94,11 @@ public CombineConstBinaryTranspose()
 
         if (y is Const)
         {
+            if (y.CheckedShape.Rank == 0)
+            {
+                return Transpose(Binary(binary.BinaryOp, x, y), perm);
```
Comment (aaltonenzhang) [inline src/Nncase.Passes/Rules/Neutral/CombineTranspose.cs:99]: please create Binary using Binary(binary.BinaryOp, x, y).InheritMetaData(binaryCall)

**Comment by xhuohai** [comment]
close this since all changes are merged by another PR.

**Pull request closed without merging**